data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostStateValidatorBalances.java

```diff
@@ -0,0 +1,86 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.handlers.v1.beacon;
+
+import static tech.pegasys.teku.beaconrestapi.BeaconRestApiTypes.PARAMETER_STATE_ID;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_BEACON;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.STRING_TYPE;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.base.Throwables;
+import java.util.List;
+import java.util.Optional;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.migrated.StateValidatorBalanceData;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+
+public class PostStateValidatorBalances extends RestApiEndpoint {
+  private final ChainDataProvider chainDataProvider;
+
+  public PostStateValidatorBalances(final DataProvider dataProvider) {
+    this(dataProvider.getChainDataProvider());
+  }
+
+  PostStateValidatorBalances(final ChainDataProvider chainDataProvider) {
+    super(
+        EndpointMetadata.post(GetStateValidatorBalances.ROUTE)
+            .operationId("postStateValidatorBalances")
+            .summary("Get validator balances from state")
+            .description("Returns filterable list of validator balances.")
+            .tags(TAG_BEACON)
+            .pathParam(PARAMETER_STATE_ID)
+            .requestBodyType(DeserializableTypeDefinition.listOf(STRING_TYPE))
+            .response(SC_OK, "Request successful", GetStateValidatorBalances.RESPONSE_TYPE)
+            .withNotFoundResponse()
+            .build());
+    this.chainDataProvider = chainDataProvider;
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final Optional<List<String>> validators;
+    try {
+      validators = request.getOptionalRequestBody();
+    } catch (RuntimeException e) {
+      final Throwable throwable = Throwables.getRootCause(e);
+      if (throwable instanceof JsonParseException) {
+        request.respondError(SC_BAD_REQUEST, throwable.getMessage());
+      } else {
+        throw e;
+      }
+      return;
```

Comment:
Same suggestion on how to handle the exception I did here: https://github.com/Consensys/teku/pull/7739#discussion_r1400020554